interpreter/core/llm/llm.py

```diff
@@ -166,17 +166,24 @@ def run(self, messages):
         ## Start forming the request
 
         params = {
-            "model": self.model,
             "messages": messages,
             "stream": True,
         }
 
+        if self.model:
+            # if model is azure, set the deployment_id and model
+            if self.model.startswith("azure/"):
+                params["deployment_id"] = self.model.split("/").pop()
+                params["model"] = params["deployment_id"]
+            # otherwise, set only the model
+            else:
+                params["model"] = self.model
+
         # Optional inputs
         if self.api_key:
             params["api_key"] = self.api_key
         if self.api_base:
             params["api_base"] = self.api_base
-            params["custom_llm_provider"] = "openai"
```

Comment:
What's the benefit of removing this?

Reply:
Thanks for the comment! I've explained the reason for this in the comments below.

https://github.com/KillianLucas/open-interpreter/pull/1026#discussion_r1501327535